libraries/AP_Networking/AP_Networking.h

```diff
@@ -0,0 +1,191 @@
+
+#pragma once
+
+#include "AP_Networking_Config.h"
+
+#if AP_NETWORKING_ENABLED
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Param/AP_Param.h>
+
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+#include "lwipthread.h"
+#include "lwip/udp.h"
+#include <lwip/ip_addr.h>
+#else
+#include <AP_Common/missing/byteswap.h>
+#define IP4_ADDR_VALUE(a,b,c,d) be32toh(UINT32_VALUE(a,b,c,d))
+#endif
+
+#define IP4_ADDR_VALUE_FROM_ARRAY(array) IP4_ADDR_VALUE(array[0],array[1],array[2],array[3])
+#define IP4_ADDR_FROM_ARRAY(dest_ip, array) IP4_ADDR(dest_ip, array[0],array[1],array[2],array[3])
+#define IP_ADDR_FROM_ARRAY(dest_ip, array) IP4_ADDR_FROM_ARRAY(dest_ip, array)
+
+class AP_Networking
+{
+public:
+    AP_Networking();
+
+    /* Do not allow copies */
+    CLASS_NO_COPY(AP_Networking);
+
+    void init();
```

Comment:
I think we should add some more comments above these methods.  Many of the names make it obvious but still, I think we've got many less comments than most libraries.

Reply:
I've added a bunch.. We could still use more though. There will be plenty of follow-up PRs expanding on all this stuff. I fact, most of them are not currently used in this PR. They're all helpers for things like scripting.